src/mpid/ch4/netmod/ucx/ucx_types.h

```diff
@@ -23,8 +23,12 @@
 /* Active Message Stuff */
 #define MPIDI_UCX_NUM_AM_BUFFERS       (64)
 #define MPIDI_UCX_MAX_AM_EAGER_SZ      (16*1024)
-#define MPIDI_UCX_TAG_USABLE_BITS      (MPIR_TAG_USABLE_BITS >> 1)
-#define MPIDI_UCX_AM_TAG               (MPIDI_UCX_TAG_USABLE_BITS + 1)
+#ifdef MPID_TAG_AUX_BITS
+#undef MPID_TAG_AUX_BITS
+#define MPID_TAG_AUX_BITS              (1)      /* we reserve one bit to differentiate AMs from native messages */
```

Comment:
@raffenet sorry I haven't been able to clearly explain my point to you earlier. The only reasonable solution I could find for UCX AM tag bit (from my perspective) is to allow the netmod to reserve some auxiliary bits for dedicated functionalities. These are accounted for by `MPID_TAG_AUX_BITS`. Most netmods just use the default (`0`), UCX redefines it to `1`. In this way UCX netmod is still using `31` bits for the tag and not `30`. Since the position of the error and collective bits is now relative to the length of the tag this it cannot be changed to `30`. What changes is the `MPIR_TAG_USABLE_BITS`, which defines `tag_ub`. This is the only variable that has to account for the AM tag in UCX, or any other tag bit reserved by any other netmod. 
The alternative solution, from my perspective, would be to change the UCX AM code to remove the AM bit from the tag.

Reply:
What I don't understand is why `ch4/ofi` doesn't need to do this. It reserves 4 bits for dedicated functionality, doesn't it?

Reply:
Actually, no. I am mixing up the OFI AM header with the tag bits so that's not the issue.

Reply:
OK, I think I understand now that OFI netmod defines these protocol bits outside of the user tag space, whereas UCX netmod does not. I think your suggestion to move the AM bit outside of the tag is the right thing to do for simplicity.

Reply:
I am not sure how much changes are required for this. Should this be a separate PR?

Reply:
I think it can be part of this PR. Moving the AM bit outside of the user tag should not be a big burden. It can be done prior to the alignment commits.